Add dwellings into the Editor

ihhub opened the pull request:
relates to #6845

![image](https://github.com/ihhub/fheroes2/assets/19829520/955a20a3-685a-4f3d-bc67-c701831dbebc)

I also added extra check for the objects.

Districh-ru commented on src/fheroes2/maps/map_object_info.cpp:
```diff
@@ -337,7 +337,326 @@ namespace
+            objects.emplace_back( std::move( object ) );
+        }
+
+        // Dragon City.
```
The passability for Dragon city is slightly different.
Original Editor:
![изображение](https://github.com/ihhub/fheroes2/assets/113276641/aa168ff0-3f00-411c-bd26-4b2813e73499)

This PR:
![изображение](https://github.com/ihhub/fheroes2/assets/113276641/6a600b5f-c0f5-4eeb-b2bb-75a868cf29cd)

Districh-ru commented on src/fheroes2/maps/map_object_info.cpp:
```diff
@@ -337,7 +337,326 @@ namespace
+            objects.emplace_back( std::move( object ) );
+        }
+
+        // Watch Tower (for Orcs).
```
It is interesting - the same objects (with and without snow) have different passabilities:
![изображение](https://github.com/ihhub/fheroes2/assets/113276641/555474c4-722f-45c0-b14b-ac86dc702dad)

The same is for the original editor. Should we choose one type of passabilities for such objects ... or leave it as is:  the showy Watch Tower has more pass restrictions because of snow around?)

ihhub replied:
Essentially, the snow object is taller leading to different passabilities. I recommend to leave it as it is as we are not changing passabilities in the engine at the moment.

Districh-ru replied:
Sure. I was just interested that initially almost the same objects have different passabilities.
I think, like a firecamp, it is taller because there was need of some space do draw the sow, so the tower sprite was lifted a bit. :)

Districh-ru commented on src/fheroes2/maps/map_object_info.cpp:
```diff
@@ -337,7 +337,326 @@ namespace
+            objects.emplace_back( std::move( object ) );
+        }
+
+        // Graveyard (for Zombies).
```
Visually the left tile is passable from the top (it has only a shadow). Should we change the passability for it?
![изображение](https://github.com/ihhub/fheroes2/assets/113276641/46b52583-ef38-44c9-b48c-b43a2b187074)

ihhub replied:
Good idea!

Districh-ru reviewed (changes requested):
Hi, @ihhub, I left some passability related comments. Could you please take a look?

ihhub commented:
Hi @Districh-ru , I addressed your comments and also fixed warning message popup to align with the latest changes there. Could you please have a look again?

Districh-ru reviewed (approved):
👍
